OutOfSchool/OutOfSchool.WebApi.Tests/Validators/CustomAgeAttributeTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using NUnit.Framework;
+using OutOfSchool.Common.Validators;
+
+namespace OutOfSchool.WebApi.Tests.Validators;
+
+[TestFixture]
+public class CustomAgeAttributeTests
+{
+    [Test]
+    public void IsValid_WhenDateIsNull_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = null as DateTime?;
+        var customDayOfBirthAttribute = new CustomAgeAttribute();
+
+        // Act
+        var isValid = customDayOfBirthAttribute.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsNotDateTimeOrDateOnly_ShouldReturnFalse()
+    {
+        // Arrange
+        var dateOfBirth = "1999-01-01";
+        var customDayOfBirthAttribute = new CustomAgeAttribute();
+
+        // Act
+        var isValid = customDayOfBirthAttribute.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsFalse(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsDateTime_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = DateTime.UtcNow;
+        var customDayOfBirthAttribute = new CustomAgeAttribute();
+
+        // Act
+        var isValid = customDayOfBirthAttribute.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenDateIsDateOnly_ShouldReturnTrue()
+    {
+        // Arrange
+        var dateOfBirth = DateOnly.FromDateTime(DateTime.UtcNow);
+        var customDayOfBirthAttribute = new CustomAgeAttribute();
+
+        // Act
+        var isValid = customDayOfBirthAttribute.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.IsTrue(isValid);
+    }
+
+    [Test]
+    public void IsValid_WhenMinAgeIsNotPositive_ShouldThrowInvalidOperationException()
+    {
+        // Arrange
+        var dateOfBirth = DateTime.UtcNow;
+        var customDayOfBirthAttribute = new CustomAgeAttribute() { MinAge = -1 };
+
+        // Act
+        TestDelegate action = () => customDayOfBirthAttribute.IsValid(dateOfBirth);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(action);
```

Comment:
(Test design) Creation of delegate is not an "Act", it's closer to "Arrange" or "Assert". As for me, I'll rewrite this as 
```
        // Assert
        Assert.Throws<InvalidOperationException>(() => customDayOfBirthAttribute.IsValid(dateOfBirth));
```

Reply:
Please check other tests as well